cpp/include/cudf/column/column_view.hpp

```diff
@@ -367,16 +398,22 @@ class column_view : public detail::column_view_base {
 
   /**
    * @brief Returns the number of child columns.
+   *
+   * @return The number of child columns
    */
   [[nodiscard]] size_type num_children() const noexcept { return _children.size(); }
 
   /**
    * @brief Returns iterator to the beginning of the ordered sequence of child column-views.
+   *
+   * @return Child column-views beginning iterator
    */
   auto child_begin() const noexcept { return _children.cbegin(); }
 
   /**
    * @brief Returns iterator to the end of the ordered sequence of child column-views.
+   *
+   * @return Child column-views end iterator
```

Comment:
For both `begin` and `end` I'm recommending language based on how [cppreference documents iterators for e.g. std::vector](https://en.cppreference.com/w/cpp/container/vector/end)
```suggestion
   * @return An iterator to a `column_view` to the element following the last child column
```